connector/google/google.go

```diff
@@ -252,34 +263,95 @@ func (c *googleConnector) createIdentity(ctx context.Context, identity connector
 	return identity, nil
 }
 
+func (c *googleConnector) getAllGroups(ctx context.Context, userKey string) ([]string, error) {
+	directMemberships, err := c.adminSrv.Groups.List().
+		UserKey(userKey).
+		Domain(c.domain).
+		Context(ctx).
+		Do()
+	if err != nil {
+		return nil, err
+	}
+
+	var groups []string
+	var groupsCh = make(chan string)
+	checkedGroups := sync.Map{}
+	g, cctx := errgroup.WithContext(ctx)
+
+	for _, group := range directMemberships.Groups {
+		email := group.Email
+		g.Go(func() error {
+			transitiveGroups, err := c.getGroups(cctx, email, &checkedGroups)
```

Comment:
The `getGroups` method returns a slice, and then this slice is sent to the channel. Can this be optimized to use a channel to return groups for the `getGroups` method also?

Reply:
I find the readability better if we avoid passing channels around and instead only use them at synchronization points. If you strongly disagree, I can change it.